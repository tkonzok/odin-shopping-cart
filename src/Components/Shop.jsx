import { useState } from "react";
import "../styles/normalize.css";
import "../styles/style.css";
import Dataset from "../assets/data.json";

function Product({ product }) {
  const [amount, setAmount] = product.stock > 0 ? useState(1) : useState(0);

  const handleClickDecrease = () => {
    if (amount > 0) setAmount(amount - 1);
  };

  const handleClickIncrease = () => {
    if (amount < product.stock) setAmount(amount + 1);
  };

  return (
    <div className="product">
      <img className="image" src={product.image}></img>
      <p className="title">{product.title}</p>
      <p className="subtitle">{product.subtitle}</p>
      <p className="published">
        {"("}
        {product.published}
        {")"}
      </p>
      <p className={product.stock == 0 ? "price sold-out" : "price"}>
        â‚¬ {product.price}
      </p>
      {product.stock > 4 && <p className="stock">Available</p>}
      {product.stock > 0 && product.stock < 5 && (
        <p className="stock limited">Less than 5 in stock</p>
      )}
      {product.stock == 0 && <p className="stock sold-out">Out of stock</p>}
      <p className="rating">
        {product.rating.rate}/5{" "}
        <span className="count">
          {"("}
          {product.rating.count}
          {")"}
        </span>
      </p>
      <div className="amount">
        <button onClick={handleClickDecrease}>-</button>
        <p>{amount}</p>
        <button onClick={handleClickIncrease}>+</button>
      </div>
      <div
        className={product.stock == 0 ? "add-to-cart sold-out" : "add-to-cart"}
      >
        <button>ADD TO CART</button>
      </div>
    </div>
  );
}

function Shop() {
  let Products = [];
  for (let i = 0; i < Dataset.data.length; i++) {
    Products.push(
      <Product key={Dataset.data[i].id} product={Dataset.data[i]} />
    );
  }

  return (
    <>
      <main>
        <section className="shop-container">
          <h2>Shop</h2>
          <div className="products-container">{Products}</div>
        </section>
      </main>
    </>
  );
}

export default Shop;
