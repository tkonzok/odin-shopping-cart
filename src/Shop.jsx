import PropTypes from "prop-types";
import { useState } from "react";
import Nav from "./Nav.jsx";
import Header from "./Header.jsx";
import Footer from "./Footer.jsx";
import "./normalize.css";
import "./style.css";

function Shop() {
  return (
    <>
      <main>
        <section className="content">
          <h2>Shop</h2>
        </section>
      </main>
    </>
  );
}

export default Shop;
