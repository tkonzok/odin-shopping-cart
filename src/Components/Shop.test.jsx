import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Shop from "./Shop.jsx";

describe("Shop component", () => {
  it("renders correct headline", () => {
    render(<Shop />);
    expect(screen.getByRole("heading").textContent).toMatch(/Shop/i);
  });

  it("renders cart 0 on load", () => {
    const { container } = render(<Shop />);
    expect(container).toMatchSnapshot();
  });

  it("renders cart 1 after button click", async () => {
    const user = userEvent.setup();

    render(<Shop />);
    const button = screen.getByRole("button", { name: "Add Item" });

    await user.click(button);

    expect(screen.getByText(/Cart:/i).textContent).toMatch(/Cart: 1/i);
    screen.debug();
  });
});
