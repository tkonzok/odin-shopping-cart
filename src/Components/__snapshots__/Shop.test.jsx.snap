// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Shop component > renders cart 0 on load 1`] = `
<div>
  <main>
    <section
      class="content"
    >
      <h2>
        Shop
      </h2>
      <p>
        Cart: 
        0
      </p>
      <button>
        Add Item
      </button>
    </section>
  </main>
</div>
`;
